services/ui-src/src/utils/auth/useUser.ts

```diff
@@ -4,10 +4,11 @@ import { UserContext } from "./userContext";
 export const useUser = (): any => {
   const context = useContext(UserContext);
 
-  if (context === undefined) {
+  if (typeof context === "undefined") {
```

Comment:
Why this change? Just wondering for my own learning. I have a preference for direct checks against undefined or loose equality check against `null` (e.g., `null == context`). But just wondering what this typeof covers over the string equality with `undefined`?

Reply:
They're very similar, but using `typeof` it won't throw an error if the variable hasn't been declared, so I always default to using `typeof` and strict equality when checking for undefined. It's an especially common pattern in SSR apps to guard server-side vs. client-side code by checking for the window obj, like:

```
if (typeof window !== 'undefined') {
  // do whatever
}
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined#typeof_operator_and_undefined